caput/mpiarray.py

```diff
@@ -251,7 +251,9 @@
 import numpy as np
 import numpy.typing as npt
 
-from caput import fileformats, misc, mpiutil
+from . import mpiutil
+from .memdata import fileformats
+from .memdata._io import open_h5py_mpi
```

Comment:
My take on private modules is: access to them should stop at their parent.  (i.e. they can be accessed by the parent, and by siblings, and by descendants of siblings (nieces?), but not from here, which is what I guess is an "aunt" in this analogy).

I would recommend either renaming `.memdata._io` to `.memdata.io` and just making it public:
```suggestion
from .memdata.io import open_h5py_mpi
```
 or else import the stuff from `.memdata._io` into memdata's `__init__.py`, to hide `.memdata`'s private modules from the outside, and and then here do:
```suggestion
from .memdata import open_h5py_mpi
```

Reply:
Yeah - I'll make these changes for now.

I envision that this I/O stuff will eventually only be used within `memdata`, or whatever it ends up being called, rather than being split between `mpiarray` and `memh5`